docs/shared/packages/webpack/webpack-config-setup.md

```diff
@@ -0,0 +1,89 @@
+# Configure webpack on your Nx workspace
+
+You can configure Webpack using a `webpack.config.js` file in your project. You can set the path to this file in your `project.json` file, in the `build` target options:
+
+```json
+//...
+"my-app": {
+    "targets": {
+        //...
+        "build": {
+            "executor": "@nrwl/webpack:webpack",
+            //...
+            "options": {
+                //...
+                "webpackConfig": "apps/my-app/webpack.config.js"
+            },
+            "configurations": {
+                ...
+            }
+        },
+    }
+}
+```
+
+In that file, you can add the necessary configuration for Webpack. You can read more on how to configure webpack in the [Webpack documentation](https://webpack.js.org/concepts/configuration/).
+
+## Using webpack with `isolatedConfig`
+
+You can set `isolatedConfig` to `true` in your `project.json` file, in the `build` target options:
+
+```json
+//...
+"my-app": {
+    "targets": {
+        //...
+        "build": {
+            "executor": "@nrwl/webpack:webpack",
+            //...
+            "options": {
+                //...
+                "webpackConfig": "apps/my-app/webpack.config.js",
+                "isolatedConfig": true
+            },
+            "configurations": {
+                ...
+            }
+        },
+    }
+}
+```
+
+This means that Nx will not apply the webpack plugins automatically. In that case, you need to manually add these plugins in your `webpack.config.js` file for Nx to work properly. Let's see how to do that.
+
+### Basic configuration for Nx
+
+You should start with a basic webpack configuration for Nx in your project, that looks like this:
+
+```ts
+// apps/my-app/webpack.config.js
+const { composePlugins, withNx } = require('@nrwl/webpack');
+
+module.exports = composePlugins(withNx(), (config) => {
+  // customize webpack config here
+  return config;
+});
+```
+
+The `withNx()` plugin adds the necessary configuration for Nx to work with Webpack. The `composePlugins` function allows you to add other plugins to the configuration.
```

Comment:
It's worth adding a note here that our webpack.config.js file is an enhanced version where the function gets two arguments:

1. Webpack config object (same as https://webpack.js.org/configuration/)
2. An Nx context, where the executor options and executor context are passed. This is useful if user needs to do different things depending on the options or context.

Reply:
(2) is not 100% clear to me. How would someone load the `context`?

Reply:
Ok!